### PR TITLE
Chore: Add missing audio types

### PR DESCRIPTION
**Pull request by michaelkoelewijn**
- Also support bluetooth devices

### CODE REVIEW

File: android/src/main/java/com/robinpowered/AudioJack/AudioJackModule.java
Comment (yinghang) [inline android/src/main/java/com/robinpowered/AudioJack/AudioJackModule.java]: Adding `TYPE_WIRED_HEADSET` LGTM but this library is meant for determining whether a wired audio device is plugged in, as described in the package name. Also, adding this means having an inconsistent API between iOS and Android since the iOS version only looks for `AVAudioSessionPortHeadphones`.

Will defer to active maintainer @christophermark

Comment (michaelkoelewijn) [reply android/src/main/java/com/robinpowered/AudioJack/AudioJackModule.java]: For now i can remove the bluetooth types in order for a merge? :)

Comment (christophermark) [reply android/src/main/java/com/robinpowered/AudioJack/AudioJackModule.java]: @yinghang is right - adding `WIRED_HEADSET` makes sense, as it keeps with the API.

If you want to detect Bluetooth types, feel free to add a separate getter method to return those (in a separate PR).

If you remove bluetooth methods, we can merge 😄

Comment (RWOverdijk) [reply android/src/main/java/com/robinpowered/AudioJack/AudioJackModule.java]: I'd maybe make this opt-in,  through a boolean argument. Something like "includeBluetoothDevices" and default to false. That way it's backwards compatible and opt-in, and you don't neglect the name of the module.

Comment (yinghang) [reply android/src/main/java/com/robinpowered/AudioJack/AudioJackModule.java]: It's Chris' call but someone will still have to implement the same thing for iOS

Comment (RWOverdijk) [reply android/src/main/java/com/robinpowered/AudioJack/AudioJackModule.java]: That's why I suggested this. That makes the transition easier once iOS gets added. I can add that actually once I implement the headphone detection for my app, I wouldn't mind.

Comment (michaelkoelewijn) [reply android/src/main/java/com/robinpowered/AudioJack/AudioJackModule.java]: @christophermark Bluetooth types have been removed!

**Comment by michaelkoelewijn** [comment]
Is this still under review or is there another reason this has not been merged?
I'll be forced to fork and use that version in my project otherwise.

**Comment by RWOverdijk** [comment]
Aaah shucks... So what's the plan for bluetooth based headphones then? New PR? 😄

**Review by christophermark — APPROVED** [review]
Sorry for the late review on this one.

👍 Thanks for the contribution - I'll cut a release with this new change.

**Comment by christophermark** [comment]
@RWOverdijk regarding Bluetooth support:

The goal is to add functionality without introducing breaking changes. Changing our method arguments can be a breaking change, but there's another way that's cleaner to introduce Bluetooth support.

I would recommend adding an additional method for checking if there's a connected bluetooth device, on iOS and Android. Then you can combine both checks to see if there's a connected device:

```js
let audioDeviceConnected = AudioJackManager.isPluggedIn() && AudioJackManager.bluetoothDeviceConnected()
```

**Comment by christophermark** [comment]
https://github.com/robinpowered/react-native-audio-jack/releases/tag/v1.0.5

Released!

**Comment by RWOverdijk** [comment]
@christophermark I know, I commented that before:

> I'd maybe make this opt-in, through a boolean argument.

To keep it BC. You never listen to me! I want a divorce :(

**Comment by christophermark** [comment]
@RWOverdijk fwiw: 

> Changing our method arguments can be a breaking change

**Comment by RWOverdijk** [comment]
Out of curiosity... How is that a breaking change?

**Comment by christophermark** [comment]
Java is a strictly typed language. If you pass parameters to a method that don't match the expected arguments, it leads to a crash. 

Either way, it's not a great separation of concerns. This function is called `isPluggedIn` and describes whether a device is physically plugged into a headphone jack. If you want to know whether any audio device is connected, that would be more suitable in a separate method.

**Comment by RWOverdijk** [comment]
I know, that's why I said it should be an optional argument so it won't crash 😄

In any case I'll just fork the module. Thanks for the fast replies, I appreciate it.

**Pull request merged**
